subsys/disk/disk_access.c

```diff
@@ -167,52 +167,47 @@ int disk_access_ioctl(const char *pdrv, uint8_t cmd, void *buf)
 
 int disk_access_register(struct disk_info *disk)
 {
-	int rc = 0;
+	k_spinlock_key_t spinlock_key;
 
-	k_mutex_lock(&mutex, K_FOREVER);
 	if ((disk == NULL) || (disk->name == NULL)) {
 		LOG_ERR("invalid disk interface!!");
-		rc = -EINVAL;
-		goto reg_err;
+		return -EINVAL;
 	}
 
 	if (disk_access_get_di(disk->name) != NULL) {
 		LOG_ERR("disk interface already registered!!");
-		rc = -EINVAL;
-		goto reg_err;
+		return -EINVAL;
 	}
 
 	/* Initialize reference count to zero */
 	disk->refcnt = 0U;
 
+	spinlock_key = k_spin_lock(&lock);
 	/*  append to the disk list */
 	sys_dlist_append(&disk_access_list, &disk->node);
 	LOG_DBG("disk interface(%s) registered", disk->name);
-reg_err:
-	k_mutex_unlock(&mutex);
-	return rc;
+	k_spin_unlock(&lock, spinlock_key);
+	return 0;
 }
 
 int disk_access_unregister(struct disk_info *disk)
 {
-	int rc = 0;
+	k_spinlock_key_t spinlock_key;
 
-	k_mutex_lock(&mutex, K_FOREVER);
 	if ((disk == NULL) || (disk->name == NULL)) {
 		LOG_ERR("invalid disk interface!!");
-		rc = -EINVAL;
-		goto unreg_err;
+		return -EINVAL;
 	}
 
 	if (disk_access_get_di(disk->name) == NULL) {
 		LOG_ERR("disk interface not registered!!");
-		rc = -EINVAL;
-		goto unreg_err;
+		return -EINVAL;
 	}
+
+	spinlock_key = k_spin_lock(&lock);
 	/* remove disk node from the list */
 	sys_dlist_remove(&disk->node);
 	LOG_DBG("disk interface(%s) unregistered", disk->name);
-unreg_err:
-	k_mutex_unlock(&mutex);
-	return rc;
+	k_spin_unlock(&lock, spinlock_key);
```

Comment:
Just a nitpick, we could move it above the LOG_DBG.

Reply:
Done